sabelotodo/routes.py

```diff
@@ -50,9 +50,41 @@ def create_item():
         db.session.rollback()
         return "Database error", 500
 
-    return jsonify(item), 200
-    # TODO: This should use item_schema.dump, but currently this leads to small
-    # discrepancies in date string format that break tests. (lbv 2020-4-2)
+    return item_schema.dumps(item), 200
+
+
+@app.route('/item/<int:itemid>', methods=["PATCH"])
+def patch_item(itemid):
+    item = Item.query.get_or_404(itemid)
+    json_data = request.get_json()
+
+    if not json_data:
+        return "No data provided", 400
+    if "id" in json_data:
+        return "ID numbers shouldn't change", 400
+
+    item_data = item_schema.dump(item)
+    item_data.update(json_data)
+
+    try:
+        item_schema.load(item_data)
+        for k, v in item_data.items():
```

Comment:
hmmmm... I feel like there ought to be a build-in way to do updates in your schema library.
Did you try something like this?
https://stackoverflow.com/questions/31891676/update-row-sqlalchemy-with-data-from-marshmallow

Reply:
The approach at that link didn't work, but I've found one that seems to.